src/app/obfboard.spec.ts

```diff
@@ -146,3 +145,132 @@ describe('obfboard.LoadBoardAction', () => {
     expect(lba).toBeTruthy();
   });
 });
+
+describe('obfboard.validation', () => {
+  it('should validate required board id', () => {
+    const input = {
+      name: 'board_name',
+      grid: {
+        rows: 2,
+        columns: 2,
+        order: [[1, null],
+        [null, 2]]
+      },
+      buttons: [],
+      images: [],
+      sounds: [],
+    };
+    try {
+      new OBFBoard().deserialize(input);
+      expect(true).toBe(false);
+    } catch (e) {
+      expect(e.errorCode).toBe(ErrorCodes.OBF_VALIDATION);
+      expect(e.message).toContain('id should not be empty');
+      expect(e.message).toContain('id must be a string');
+    }
+  });
+
+  it('should validate grid array size - rows', () => {
+    const input = {
+      id: 'test',
+      name: 'board_name',
+      grid: {
+        rows: 3,
+        columns: 2,
+        order: [
+          [1, null],
+          [null, 2]
+        ]
+      },
+      buttons: [],
+      images: [],
+      sounds: [],
+    };
+    try {
+      new OBFBoard().deserialize(input);
+      expect(true).toBe(false);
+    } catch (e) {
+      expect(e.errorCode).toBe(ErrorCodes.OBF_VALIDATION);
+      expect(e.message).toContain('Grid has 2 rows but should have 3');
+    }
+  });
+
+  it('should validate grid array size - columns', () => {
+    const input = {
+      id: 'test',
+      name: 'board_name',
+      grid: {
+        rows: 2,
+        columns: 2,
+        order: [
+          [1, null],
+          [null, 2, 3]
+        ]
+      },
+      buttons: [],
+      images: [],
+      sounds: [] };
```

Comment:
Not now for this review, but would it be useful to have a standard empty board json that could be used and edited per test as needed?

Reply:
Hmmmm dunno.  How would we amend it for a specific test?

Reply:
let input = basicEmptyBoardJson();
input.grid.order = [["2", null]], [null, "1", null"]];